src/main/java/ua/tc/marketplace/controller/AttributeController.java

```diff
@@ -0,0 +1,70 @@
+package ua.tc.marketplace.controller;
+
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import ua.tc.marketplace.model.dto.attribute.AttributeDto;
+import ua.tc.marketplace.model.dto.attribute.CreateAttributeDTO;
+import ua.tc.marketplace.model.dto.attribute.UpdateAttributeDTO;
+import ua.tc.marketplace.service.AttributeService;
+
+import java.util.List;
+
+/**
+ * AttributeController handles HTTP requests related to CRUD operations for the Attribute entity. It
+ * provides an API for retrieving, creating, updating, and deleting attributes.
+ */
+@RestController
+@RequestMapping("/api/v1/attribute")
+@RequiredArgsConstructor
+@Slf4j
+public class AttributeController {
+
+    private final AttributeService attributeService;
+
+    @GetMapping
+    public ResponseEntity<List<AttributeDto>> getAllAttributes(Pageable pageable) {
+        log.info("Request to get all attributes");
+        List<AttributeDto> attributes = attributeService.findAll(pageable);
+        log.info("Attributes were retrieved successfully");
+        return ResponseEntity.ok(attributes);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<AttributeDto> getAttributeById(@PathVariable Long id) {
+        log.info("Request to get attribute with ID: {}", id);
+        AttributeDto attributeDto = attributeService.findById(id);
+        log.info("Attribute with ID: {} was retrieved successfully", id);
+        return ResponseEntity.ok(attributeDto);
+    }
+
+    @PostMapping
+    public ResponseEntity<AttributeDto> createAttribute(@Valid @RequestBody CreateAttributeDTO attributeDTO) {
```

Comment:
You're using different styles for abbreviations in class and variables names (AttributeDto and CreateAttributeDTO). Even though both are correct, you need to choose one. All previosly created Dtos has this style (AttributeDto).

Reply:
actually I dont understand what do you mean))) I looked at the user dto package and it has the same name structure that are here in attributes))

But i found another questionable place: there are two exactly the same attribute dto with different names (CreateAttributeDto and UpdateAttributeDto). In some cases it is possible to make like this. But in our case i renamed one of them into `AttributeRequest` and deleted other one